cli/azd/pkg/infra/provisioning/manager.go

```diff
@@ -6,101 +6,80 @@ package provisioning
 )
 
 // Manages the orchestration of infrastructure provisioning
 type Manager struct {
-	azCli              azcli.AzCli
-	env                *environment.Environment
-	provider           Provider
-	writer             io.Writer
-	console            input.Console
-	accountManager     account.Manager
-	userProfileService *azcli.UserProfileService
-	subResolver        account.SubscriptionTenantResolver
-	interactive        bool
+	serviceLocator      ioc.ServiceLocator
+	env                 *environment.Environment
+	console             input.Console
+	provider            Provider
+	alphaFeatureManager *alpha.FeatureManager
+	projectPath         string
+	options             *Options
+}
+
+func (m *Manager) Initialize(ctx context.Context, projectPath string, options Options) error {
+	m.projectPath = projectPath
+	m.options = &options
+
+	provider, err := m.newProvider(ctx)
+	if err != nil {
+		return fmt.Errorf("initializing infrastructure provider: %w", err)
+	}
+
+	m.provider = provider
+	return m.provider.Initialize(ctx, projectPath, options)
 }
 
 // Prepares for an infrastructure provision operation
 func (m *Manager) Plan(ctx context.Context) (*DeploymentPlan, error) {
-	deploymentPlan, err := m.plan(ctx)
+	deploymentPlan, err := m.provider.Plan(ctx)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("planning infrastructure provisioning: %w", err)
 	}
 
 	return deploymentPlan, nil
 }
 
 // Gets the latest deployment details for the specified scope
 func (m *Manager) State(ctx context.Context) (*StateResult, error) {
-	var stateResult *StateResult
-
-	err := m.runAction(
-		ctx,
-		"Retrieving Infrastructure State",
-		m.interactive,
-		func(ctx context.Context, spinner *spin.Spinner) error {
-			queryTask := m.provider.State(ctx)
-
-			go func() {
-				for progress := range queryTask.Progress() {
-					m.updateSpinnerTitle(spinner, progress.Message)
-				}
-			}()
-
-			go m.monitorInteraction(spinner, queryTask.Interactive())
-
-			result, err := queryTask.Await()
-			if err != nil {
-				return err
-			}
-
-			stateResult = result
-
-			return nil
-		},
-	)
-
+	result, err := m.provider.State(ctx)
 	if err != nil {
 		return nil, fmt.Errorf("error retrieving state: %w", err)
 	}
 
-	return stateResult, nil
+	return result, nil
 }
 
 // Deploys the Azure infrastructure for the specified project
 func (m *Manager) Deploy(ctx context.Context, plan *DeploymentPlan) (*DeployResult, error) {
 	// Apply the infrastructure deployment
-	deployResult, err := m.deploy(ctx, plan)
+	deployResult, err := m.provider.Deploy(ctx, plan)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("error deploying infrastructure: %w", err)
 	}
 
 	if err := UpdateEnvironment(m.env, deployResult.Deployment.Outputs); err != nil {
 		return nil, fmt.Errorf("updating environment with deployment outputs: %w", err)
 	}
 
+	// make sure any spinner is stopped
+	m.console.StopSpinner(ctx, "", input.StepDone)
```

Comment:
How did you know you had to add a call here do this?  Should we be doing it at the end of other long running operations, like `Destroy`?

Is it bad that we don't call this in cases where Deploy or UpdateEnvironment return a non nil error and early exit?

Reply:
The deleted private `deploy` method had this line in it. Now we are calling direct into the providers implementation.  Added it here to make it the same as before.